components/gif_picker/components/SearchGrid/SearchGrid.tsx

```diff
@@ -34,27 +36,162 @@ const mapDispatchToProps = ({
     saveSearchScrollPosition,
 });
 
-export class SearchGrid extends PureComponent {
-    static propTypes = {
-        appProps: PropTypes.object,
-        gifs: PropTypes.object,
-        resultsByTerm: PropTypes.object,
-        containerClassName: PropTypes.string,
-        keyword: PropTypes.string, // searchText, tagName
-        handleItemClick: PropTypes.func,
-        loadMore: PropTypes.func,
-        numberOfColumns: PropTypes.number,
-        scrollPosition: PropTypes.number,
-        saveSearchScrollPosition: PropTypes.func,
-    }
+type Props = {
+    appProps: Record<string, any>;
+    gifs: Record<string, Gif>;
+    resultsByTerm: Record<string, Record<string, any>>;
+    containerClassName?: string;
+    keyword: string; // searchText, tagName
+    handleItemClick: (gif: Gif) => void;
+    loadMore: () => void;
+    saveSearchScrollPosition: (scrollPosition: number) => (dispatch: DispatchFunc) => void;
+    scrollPosition: number;
+}
+
+type State = {
+    containerWidth: null | number;
+}
+
+type Gif = {
+    anonymous: boolean;
+    avgColor: string;
+    content_urls: {
```

Comment:
this type seems like it can be broken into simpler pieces to ease reading about it, at least the `content_urls` property.